speech/google/cloud/speech/client.py

```diff
@@ -46,26 +46,17 @@ class Client(client_module.Client):
 
     _connection_class = Connection
 
-    def async_recognize(self, content, source_uri, encoding, sample_rate,
+    def async_recognize(self, encoding, sample_rate, content=None,
```

Comment:
```python
class Sample(object):

    def __init__(self, content=None, source_uri=None,
                 encoding=None, sample_rate=None, client=None):
        if ((content is None and source_uri is None) or
            (content is not None and source_uri is not None)):
            raise ValueError("Supply one of 'content' or 'source_uri')
        if isinstance(content, six.text_type):
            with open(content, 'rb') as opened:
                content = content.read()
        self.content = content
        self.source_uri = source_uri
        if content is not None and sample_rate is None:
            sample_rate = self._sniff_sample_rate(content)
        self.sample_rate = sample_rate
        self.encoding = encoding or client.default_encoding
```


The implementation of `_sniff_sample_rate` is left as an exercise for the reader. ;)